src/main/scala/diplomaticobjectmodel/logicaltree/OMLogicalTree.scala

```diff
@@ -0,0 +1,85 @@
+// See LICENSE.SiFive for license details.
+
+package diplomaticobjectmodel.logicaltree
+
+import freechips.rocketchip.diplomacy.ResourceBindingsMap
+import freechips.rocketchip.diplomaticobjectmodel.model.OMComponent
+import scala.collection.mutable.ArrayBuffer
+
+
+trait LogicalTree {
+  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent]): Seq[OMComponent]
+}
+
+/** LogicalTreeEdges hold LogicalTree Nodes which will be used to construct the logical tree.
+  * The LogicalTreeEdge is used to construct a parent child relationship between different modules.
+  *
+  * 1. First a list of   val edges = ArrayBuffer[LogicalTreeEdge]() is constructed.
+  *
+  * 2. Then def getTreeMap(): Map[LogicalTree, List[LogicalTree]] converts the list into a map which represents the tree.
+  *
+  * 3. Then the tree map is converted into a OMLogicalTree by the  def makeTree(): Tree[LogicalTree].
+  *
+  * @param parent The parent LogicalTree node
+  * @param child The child LogicalTree node
+  */
+case class LogicalTreeEdge(
+  parent: LogicalTree,
+  child: LogicalTree
+)
+
+object LogicalModuleTree {
+  val edges = ArrayBuffer[LogicalTreeEdge]()
+
+  def add(parent: LogicalTree, child: LogicalTree): Unit = {
+     edges += LogicalTreeEdge(parent, child)
+  }
+
+  /**
+    * The getTreeMap function constructs a tree by converting the list of LogicalTreeEdges into a Map which
+    * contains each of the parents and their children
+    *
+    * @return
+    */
+  def getTreeMap(): Map[LogicalTree, List[LogicalTree]] = {
+    edges.groupBy(_.parent).map{ case (k, v) => (k, v.map(_.child).toList)}
```

Comment:
Does this actually add all the children to the parent? Seems like it will produce a single key per parent, with a list holdings a single child, whichever one happened to be last in the ArrayBuffer